worker/deployer/unit_manifolds.go

```diff
@@ -135,11 +135,9 @@ func UnitManifolds(config UnitManifoldsConfig) dependency.Manifolds {
 		// API for uploading and downloading charms. It provides a
 		// S3-compatible API.
 		s3CallerName: s3caller.Manifold(s3caller.ManifoldConfig{
-			AgentName:            agentName,
 			APIConfigWatcherName: apiConfigWatcherName,
 			APICallerName:        apiCallerName,
 			NewS3Client:          s3caller.NewS3Client,
-			Filter:               connectFilter,
```

Comment:
I think this is actually required.

Reply:
but it's never used on the `s3caller`, why is it required?